MSBuild fails on Linux/Mono with TypeLoadException

## Steps to reproduce

1. Build MSBuild in the `Debug-MONO` configuration and copy it to an Ubuntu 14.04 box with Mono 3.12.1 from the official Xamarin mono-devel package.
2. Clone https://github.com/mono/coreclr.git
3. Run `mono {path to msbuild} {path to coreclr}/src/build.proj -nologo -p:OS=Unix`

## Expected result

Something approximating a build

## Actual result

The following crash occurred:

```
Invalid type Microsoft.Build.Execution.BuildManager for instance field Microsoft.Build.Execution.BuildSubmission:<BuildManager>k__BackingField
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.TypeLoadException: A type load exception has occurred.
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x00000] in <filename unknown>:0
This is an unhandled exception in MSBuild Engine -- PLEASE OPEN A BUG AGAINST THE MSBUILD TEAM.
System.TypeLoadException: A type load exception has occurred.
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x00000] in <filename unknown>:0

Unhandled Exception:
System.TypeLoadException: A type load exception has occurred.
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x00000] in <filename unknown>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeLoadException: A type load exception has occurred.
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x00000] in <filename unknown>:0
vagrant@vagrant-ubuntu-trusty-64:/vagrant/git/msbuild/bin/Windows_NT/Debug-MONO$
```

## Additional notes

Through further investigation, I discovered that the `TypeLoadException` is the result of not being able to load TPL dataflow.

I was able to fix this issue by finding the dependency in `Microsoft.Build` and setting `Copy Local` to `True` in the properties pane, but I don't know if this is the *right* solution or not.

--- a/src/XMakeBuildEngine/Microsoft.Build.csproj
+++ b/src/XMakeBuildEngine/Microsoft.Build.csproj
@@ -653,6 +653,7 @@
     <Reference Include="System.Threading.Tasks.Dataflow, Version=4.5.24.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\packages\Microsoft.Tpl.Dataflow.4.5.24\lib\portable-net45+win8+wpa81\System.Threading.Tasks.Dataflow.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <!-- If TargetRetailBuildFramework is set to true, reference the public key version. Otherwise reference the project. -->
     <Reference Condition=" '$(TargetRetailBuildFramework)' == 'true' " Include="Microsoft.Build.Framework, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" />

